test/cypress/integration/settings/mask-moto-card-number-and-security-code.cy.test.js

```diff
@@ -0,0 +1,167 @@
+    let stubs = []
+    let user
+
+    if (opts.readonly) {
+      const role = {
+        permissions: [
+          {
+            name: 'transactions-details:read',
+            description: 'ViewTransactionsOnly'
+          },
+          {
+            name: 'moto-mask-input:read',
+            description: 'ViewMaskMotoOnly'
+          }
+        ]
+      }
+      user = userStubs.getUserSuccess({ userExternalId, gatewayAccountId, serviceName, role })
+    } else {
+      user = userStubs.getUserSuccess({ userExternalId, gatewayAccountId, serviceName })
+    }
+
+    const gatewayAccount = gatewayAccountStubs.getGatewayAccountSuccess({ gatewayAccountId, paymentProvider: opts.gateway, allowMoto: opts.allowMoto, motoMaskCardNumber: opts.motoMaskCardNumber, motoMaskSecurityCode: opts.motoMaskSecurityCode })
+    const card = gatewayAccountStubs.getAcceptedCardTypesSuccess({ gatewayAccountId, updated: false, maestro: opts.maestro })
+    const patchUpdateMotoMaskCardNumber = gatewayAccountStubs.patchUpdateMotoMaskCardNumber({ motoMaskCardNumber: opts.motoMaskCardNumber })
+    const patchUpdateMotoMaskSecurityCode = gatewayAccountStubs.patchUpdateMotoMaskSecurityCode({ motoMaskCardSecurityCode: opts.motoMaskCardSecurityCode })
+
+    stubs.push(user, gatewayAccount, card, patchUpdateMotoMaskCardNumber, patchUpdateMotoMaskSecurityCode)
+
+    cy.task('setupStubs', stubs)
+  }
+
+  beforeEach(() => {
+    Cypress.Cookies.preserveOnce('session', 'gateway_account')
+  })
+
+  describe('When accessing MOTO mask settings', () => {
+    describe('when gateway account has allowMoto=false', () => {
+      beforeEach(() => {
+        setupMotoStubs({ allowMoto: false })
+      })
+
+      it('should not show mask security section', () => {
+        cy.setEncryptedCookies(userExternalId, gatewayAccountId)
+        cy.visit('/settings')
+        cy.get('#moto-mask-security-settings-heading').should('not.exist')
+      })
+    })
+
+    describe('mask card number - when user has read permission and card number mask disabled', () => {
+      beforeEach(() => {
+        setupMotoStubs({ readonly: true, allowMoto: true, motoMaskCardNumber: false })
+      })
+
+      it('should show radios as disabled and card number mask disabled', () => {
+        cy.visit('/settings')
+        cy.get('.govuk-summary-list__key').eq(4).should('contain', 'Hide card number')
+        cy.get('.govuk-summary-list__value').eq(4).should('contain', 'Off')
+        cy.get('.govuk-summary-list__actions a').eq(4).contains('View')
+        cy.get('.govuk-summary-list__actions a').eq(4).click()
+        cy.title().should('eq', `MOTO - hide card numbers for ${serviceName} - GOV.UK Pay`)
+        cy.get('.pay-info-warning-box').should('exist')
+        cy.get('input[value="on"]').should('be.disabled')
+        cy.get('input[value="off"]').should('be.disabled')
+        cy.get('input[value="on"]').should('not.be.checked')
+        cy.get('input[value="off"]').should('be.checked')
```

Comment:
Don't understand why there are 4 inputs here. The screenshot attached to the story shows two radio buttons, one for "Hidden", one for "Visible"